include/click/glue.hh

```diff
@@ -388,6 +404,38 @@ click_current_processor()
 #endif
 }
 
+inline unsigned
+click_current_cpu_id()
+{
+#if !HAVE_MULTITHREAD
+    return 0;
+#elif CLICK_USERLEVEL
+#  if HAVE___THREAD_STORAGE_CLASS
+    return click_current_thread_id & 0xffff;
+#  else
+    return sched_getcpu();
```

Comment:
This seems very dangerous to me. sched_getcpu() isn't guaranteed to be < click_max_cpu_ids().